ste/jobStatusManager.go

```diff
@@ -27,34 +27,64 @@ import (
 )
 
 type JobPartCreatedMsg struct {
-	TotalTransfers uint32
-	IsFinalPart    bool
+	TotalTransfers       uint32
+	IsFinalPart          bool
 	TotalBytesEnumerated uint64
-	FileTransfers  uint32
-	FolderTransfer uint32
+	FileTransfers        uint32
+	FolderTransfer       uint32
 }
 
 type xferDoneMsg = common.TransferDetail
 type jobStatusManager struct {
-	js          common.ListJobSummaryResponse
-	respChan    chan common.ListJobSummaryResponse
-	listReq     chan bool
-	partCreated chan JobPartCreatedMsg
-	xferDone    chan xferDoneMsg
+	js              common.ListJobSummaryResponse
+	respChan        chan common.ListJobSummaryResponse
+	listReq         chan struct{}
+	partCreated     chan JobPartCreatedMsg
+	xferDone        chan xferDoneMsg
+	xferDoneDrained chan struct{}
```

Comment:
It would greatly help if we could add a comment explaining that these 2 new variables have been added in order to fix the race condition. Otherwise, it does not look very obvious to me